internal/terminal/table_test.go

```diff
@@ -0,0 +1,264 @@
+		},
+		{
+			description: "Should return correctly formatted values in the table",
+			header:      []string{"arrays", "floats", "ints", "maps/objects", "strings"},
+			data: []map[string]interface{}{
+				{
+					"ints":    1,
+					"floats":  12.34,
+					"strings": "tester string",
+					"arrays":  []string{"1", "test", "this"},
+					"maps/objects": struct {
+						test int
+						this float64
+					}{test: 1234, this: 12.345},
+				},
+				{
+					"ints":    000000001,
+					"floats":  12.34,
+					"strings": "test",
+					"arrays":  []int{1, 2, 3, 4},
+					"maps/objects": map[string]int{
+						"test": 1,
+						"this": 2,
+					},
+				},
+				{
+					"ints":    -0002,
+					"floats":  0000123.340000,
+					"strings": "hello",
+					"arrays":  []float64{1, 2.3, 4.5555555555, 6},
+					"maps/objects": map[string][]int{
+						"what happens": {1, 2, 3},
+					},
+				},
+			},
+			expectedMessage: "\n" + fmt.Sprintf("%s                  %s  %s  %s               %s      ",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("arrays"),
+					color.New(color.Bold).SprintFunc()("floats"),
+					color.New(color.Bold).SprintFunc()("ints"),
+					color.New(color.Bold).SprintFunc()("maps/objects"),
+					color.New(color.Bold).SprintFunc()("strings"),
+				}...,
+			) + `
+----------------------  ------  ----  -------------------------  -------------
+[1 test this]           12.34   1     {1234 12.345}              tester string
+[1 2 3 4]               12.34   1     map[test:1 this:2]         test         
+[1 2.3 4.5555555555 6]  123.34  -2    map[what happens:[1 2 3]]  hello        `,
+		},
+	} {
+		t.Run(tc.description, func(*testing.T) {
+			table := newTable(tc.header, tc.data)
+			message, err := table.Message()
+			if err != nil {
+				assert.Equal(t, err.Error(), tc.expectedError)
+			} else {
+				assert.Equal(t, "", tc.expectedError)
+			}
+			assert.Equal(t, message, tc.expectedMessage)
+		})
+	}
+}
+
+func TestTablePayload(t *testing.T) {
+	t.Run("Payload should return an error with a table without a header", func(t *testing.T) {
+		data := []map[string]interface{}{
+			{"test": 123, "this": "456", "data": []string{"7", "8 9", "10!"}},
+		}
+		table := newTable(nil, data)
+		payloadKeys, payloadData, err := table.Payload()
+		assert.Nil(t, payloadKeys)
+		assert.Nil(t, payloadData)
+		assert.Equal(t, err.Error(), "cannot create a table without headers")
+	})
+
+	t.Run("Payload should work with a valid table", func(t *testing.T) {
+		header := []string{"test", "this", "data"}
+		data := []map[string]interface{}{
+			{"test": 123, "this": "456", "data": []string{"7", "8 9", "10!"}},
+		}
+		table := newTable(header, data)
+		payloadKeys, payloadData, err := table.Payload()
+		assert.Nil(t, err)
+		assert.Equal(t, payloadKeys, []string{logFieldData, logFieldHeaders})
+		assert.Equal(t, payloadData[logFieldHeaders], header)
+
+		dataStr := []map[string]string{
```

Comment:
does this still want to be named `dataStr`?  I think you could go `expectedData` here, or just `data` (which would necessitate you to inline `header` and `data` above into the `newTable` constructor)